src/pages/index.js

```diff
@@ -0,0 +1,181 @@
+editFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(validationSettings, profileAddForm);
+addFormValidator.enableValidation();
+
+// Section for cards
+const cardSection = new Section(
+  {
+    renderer: (cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+    },
+  },
+  ".cards__list"
+);
+
+// Current User ID
+let currentUserId;
+
+// Initial Data Load
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    currentUserId = userData._id;
+    userInfo.setUserInfo({
+      name: userData.name,
+      about: userData.about,
+      avatar: userData.avatar,
+    });
+    cardSection.renderItems(
+      cards.map((card) => ({
+        ...card,
+        currentUserId,
+      }))
+    );
+  })
+  .catch(console.error);
+
+// Event Listeners
+profileEditButton.addEventListener("click", () => {
+  const { name, about } = userInfo.getUserInfo();
+  profileTitleInput.value = name;
+  profileDescriptionInput.value = about;
+  profileEditPopup.open();
+});
+
+profileAddButton.addEventListener("click", () => {
+  profileAddPopup.open();
+});
+
+avatarEditButton.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
+// Functions
+function createCard(cardData) {
+  const card = new Card({
+    data: cardData,
+    cardSelector: "#card-template",
+
+    handleImageClick: (name, link) => imagePopup.open(name, link),
+    handleDeleteClick: (card) => {
+      console.log("Delete button clicked");
+      confirmationPopup.open();
+
+      confirmationPopup.setSubmitAction(() => {
+        console.log(card);
+        api
+          .deleteCard(card.getId())
+          .then(() => {
+            card.removeCard();
+            confirmationPopup.close();
+          })
+          .catch(console.error)
+          .finally(() => confirmationPopup.setLoadingState(false, "delete"));
+      });
+    },
+    handleLikeClick: (isLiked) => {
+      return isLiked ? api.unlikeCard(cardData._id) : api.likeCard(cardData._id);
+    },
+    currentUserId,
+  });
+  return card.getView();
+}
+
+function handleProfileEditSubmit(formData) {
+  
+  profileEditPopup.setLoadingState(true);
+  api
+    .setUserInfo({
+      name: formData.title,
+      about: formData.description,
+    })
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        about: userData.about,
+      });
+      profileEditPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => profileEditPopup.setLoadingState(false));
+}
+
+function handleCardAddSubmit(formData) {
+  profileAddPopup.setLoadingState(true);
+  api
+    .addCard({
+      name: formData.title,
+      link: formData.url,
+    })
+    .then((cardData) => {
+      const cardElement = createCard({
+        ...cardData,
+        currentUserId,
+      });
+      cardSection.addItem(cardElement);
+      profileAddPopup.close();
+      profileAddForm.reset();
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.


```js

      profileAddForm.reset();
      addFormValidator.toggleButtonState()  // this will disable the button after resetting the form
```